extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -744,6 +792,47 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching a new creative and rendering it. This leaves
+   * the current creative displayed until the next one is ready.
+   *
+   * @param {function()} refreshEndCallback When called, this function will
+   *   restart the refresh cycle.
+   */
+  refresh(refreshEndCallback) {
+    this.isRefreshing = true;
+    this.tearDownSlot();
+    this.initiateAdRequest();
+    const promiseId = this.promiseId_;
+    this.adPromise_.then(() => {
+      if (!this.isRefreshing || promiseId != this.promiseId_) {
+        // If this refresh cycle was canceled, such as in a no-content
+        // response case, keep showing the old creative.
+        refreshEndCallback();
+        return;
+      }
+      this.togglePlaceholder(true);
```

Comment:
Also, it appears that forcing the scheduler to run earlier removes the ad loader, so moving everything into layoutCallback has the effect of essentially never showing the loader, just one ad quickly transitioning to another.

This happens with my code as:
```
  /** @override */
  layoutCallback() {
    if (this.isRefreshing) {
      this.togglePlaceholder(true);
      this.destroyFrame(true);
    }
    return this.attemptToRenderCreative().then(result => {
      if (this.isRefreshing) {
        // We don't want the next creative to appear too suddenly, so we
        // show the loader for a quarter of a second before switching to
        // the new creative.
        timerFor(this.win).delay(() => {
          this.togglePlaceholder(false);
          this.refreshReadyPromiseResolver_(result);
        }, 1000);
      }
      return result;
    });
  }
```
At some point, very quickly, after displaying the loader, the runtime removes it before `this.togglePlaceholder(false)` is called. Quick enough that you can't see it.